rust-runtime/aws-smithy-http-server/src/plugin/filter.rs

```diff
@@ -68,3 +65,61 @@ where
         either_plugin.apply(svc)
     }
 }
+
+/// Filters the application of an inner [`Plugin`] using a predicate over the
+/// [`ServiceShape::Operations`](crate::service::ServiceShape::Operations).
+///
+/// See [`filter_by_operation`] for more details.
+pub struct FilterByOperation<Inner, F> {
+    inner: Inner,
+    predicate: F,
+}
+
+impl<Ser, Op, S, Inner, F> Plugin<Ser, Op, S> for FilterByOperation<Inner, F>
```

Comment:
Hmm not a compelling use case, since it does the exact same thing as `FilterByOperationId`. I think we should only have one of them.

What useful things can we do with the operations enum?

Reply:
Right! 🤦 We should call that out in the docs.

Is there a use case for keeping `FilterByOperationId` then?

What other useful things does this operations enum enable? I can only come up with conditional application of a plugin based on the operation being invoked.

Reply:
> Is there a use case for keeping `FilterByOperationId` then?

Someone might want to filter on the namespace or perform some action on `&'static str`. This can be subsumed by generating a `fn id(&self) -> ShapeId` on the generated enum though.

Reply:
Nobody is going to filter on namespace and there's not really a need to filter specifically on `&'static str`, an enum variant serves the same filtering purpose.

What I'm trying to get at is that "filtering by operation" is, in essence, all this enables. Which is fine and definitely needed, but then there isn't really a need to additionally have `FilterByOperationId`.